File: Rest_api/models/case.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Sensor struct {
	Accelerometer []float64 `json:"accelerometer" bson:"accelerometer"`
	Gyroscope     []float64 `json:"gyroscope" bson:"gyroscope"`
	Sound         int       `json:"sound" bson:"sound"`
}

type Location struct {
	Type        string    `json:"type" bson:"type"`
	Coordinates []float64 `json:"coordinates" bson:"coordinates"`
}

type Point struct {
	ID       primitive.ObjectID `json:"id" bson:"_id"`
	Name     string             `json:"name" bson:"name"`
	PhNo     string             `json:"number"  bson:"number"`
	Loc      Location           `json:"location" bson:"location"`
	Endpoint string             `json:"serveraddress" bson:"serveraddress"`
}

type AlertMssg struct {
	Id        primitive.ObjectID `json:"id" bson:"_id"`
	Status    string             `json:"status" bson:"status"`
	DateTIme  string             `json:"datetime" bson:"datetime"`
	People    int                `json:"numbers" bson:"numbers"`
	Hospital  Point              `json:"hospital" bson:"hospital"`
	VehicleID string             `json:"vehicle_id" bson:"vehicle_id"`
	Loc       []float64          `json:"accidentlocation" bson:"accidentlocation"`
	Sensors   Sensor             `json:"sensor" bson:"sensor"`
}

File: front-end-flutter/lib/class/case.dart
class Case {
  String id;
  String status;
  String dateTime;
  int numbers;
  String vehicleId;
  Hospital hospital;
  List<dynamic> accidentLoc;
  // bool isPressed; //if the user is pressed the message or not

  Case({
    this.id,
    this.status,
    this.dateTime,
    this.numbers,
    this.vehicleId,
    this.hospital,
    this.accidentLoc,
  });

  factory Case.fromJson(Map<String, dynamic> responsevalue) {
    return Case(
        hospital: Hospital.fromJson(responsevalue['hospital']),
        id: responsevalue['id'] as String,
        status: responsevalue['status'],
        dateTime: responsevalue['datetime'],
        numbers: responsevalue['numbers'],
        vehicleId: responsevalue['vehicle_id'],
        accidentLoc: responsevalue['accidentlocation']);
  }
}

class Hospital {
  String name;
  List<dynamic> coordinates;
  String phNo;

  Hospital({this.name, this.coordinates, this.phNo});

  factory Hospital.fromJson(Map<String, dynamic> value) {
    return Hospital(
        name: value['name'] as String,
        coordinates: value['location']['coordinates'] as List<dynamic>,
        phNo: value['number'] as String);
  }
}
